### PR TITLE
Greatly refactor bin2h

### PR DESCRIPTION
**Pull request by GabrielRavier**
This basically reworks the whole control-flow of the program to make it look better.
It also adds a few bug checks for a few things (arguments, malloc, fopen, fread)

### CODE REVIEW

File: src/misc/bin2h.c [hunk 1]
```diff
@@ -1,92 +1,187 @@
...
+	size = ftell(fp);
+	rewind(fp);
+	return size;
+}
+
+// Display usage message and exit
+static void exitWithUsage()
 {
-	int result = 0;
+	errorExit(stderr, "\nUsage:\n\t%s [infile] [outfile]\n", gProcessName);
```
Comment (Clownacy) [inline src/misc/bin2h.c]: Isn't that 'stderr' not meant to be there?

Comment (GabrielRavier) [reply src/misc/bin2h.c]: Oops, fixing that...

Comment (GabrielRavier) [reply src/misc/bin2h.c]: Fixed

**Comment by GabrielRavier** [comment]
I fixed the review btw

**Comment by Clownacy** [comment]
Is there any way to squash these commits down to remove all the merges? I don't want to add hundreds of empty commits to the log.

**Comment by GabrielRavier** [comment]
Gonna reopen this as a PR into portable

**Pull request closed without merging**